app/api/utility.py

```diff
@@ -118,9 +121,8 @@ def create_query(
 
     query_template = f"""
     {DEFAULT_CONTEXT}
+    {"SELECT ?dataset ?dataset_name ?sub_id ?file_path ?image_modal WHERE {{{{" if return_agg else ""}
```

Comment:
I think this is a bit mind-bendy. I believe a slightly clearer version would be to build the template in 2 steps:

1. Build the inner query that returns the subject level data
2. IF `return_agg`, then wrap it in the outer query (prepend with the SELECT, append the GROUPBY)
3. ELSE: just return the subject-level template.

In the interest of time, I would push that to the next PR with an issue, but I think we should keep these two steps separated and not use inline if-statements in the f-string.